pkg/server/metricsloader.go

```diff
@@ -74,7 +74,7 @@ func (db *MetricsDB) Update(ctx context.Context, params *MetricsLoadParams) erro
 			// Technically a race as we could squash changes created by another update
 			// but not a big deal if that happens.
 			metrics, err := db.GetAllowedMetrics(app)
-			if err != nil {
+			if err == nil {
```

Comment:
Just wanna make sure the logic change here is as expected.

Reply:
Yeah, caught that one in testing. I may reorganize that to be more gothonic